### PR TITLE
feat: update populate data csv mgt command

### PR DESCRIPTION
**Pull request by DawoudSheraz**
### Description

- Update mgt command to use Product API as a source of truth
- Partially filled CSV is now optional. If product info is not present in CSV, the flow is not skipped for that product
- Updated the output CSV by adding new headers and new data of product API

### Todos

- Update the command in a followup PR once new changes/fields are added in API

### CODE REVIEW

File: course_discovery/apps/course_metadata/management/commands/populate_executive_education_data_csv.py [hunk 1]
```diff
@@ -98,20 +106,30 @@ def handle(self, *args, **options):
...
-                        row['title']
-                    ))
-                    continue
-                output_dict = self.get_transformed_data(row, product[0])
+            for product in products:
+                if input_reader:
+                    row = [row_item for row_item in input_reader if product['name'] == row_item['Title']]
+                    if not row or len(row) != 1:
+                        self.messages_list.append(
+                            "[MISSING PRODUCT IN CSV] Unable to find product details for product {} in CSV".format(
```
Comment (azanbinzahid) [inline course_discovery/apps/course_metadata/management/commands/populate_executive_education_data_csv.py]: can have this as constant somewhere else.

File: course_discovery/apps/course_metadata/management/commands/populate_executive_education_data_csv.py [hunk 2]
```diff
@@ -199,14 +217,22 @@ def get_transformed_data(self, partially_filled_csv_dict, product_dict):
         Returns the final representation of the data row using partially filled dict
         and the product dict.
         """
-        # TODO: To use util method once the changes are merged
-        minimum_effort, maximum_effort = 7, 10
+        try:
+            minimum_effort, maximum_effort = utils.format_effort_info(product_dict['effort'])
+        except Exception:  # pylint: disable=broad-except
+            # Exception is raised for the cases where the effort field has value in Spanish
```
Comment (azanbinzahid) [inline course_discovery/apps/course_metadata/management/commands/populate_executive_education_data_csv.py:221]: is it a TODO?

File: course_discovery/apps/course_metadata/management/commands/populate_executive_education_data_csv.py [hunk 3]
```diff
@@ -199,14 +217,22 @@ def get_transformed_data(self, partially_filled_csv_dict, product_dict):
...
+            ))
+            minimum_effort, maximum_effort = 1, 2
 
         language = self.LANGUAGE_MAP.get(product_dict['language'], 'English - United States')
 
         default_values = {  # the values that will be part of every output row in any case
-            'course_enrollment_track': 'Executive Education',
-            'course_run_enrollment_track': 'Executive Education',
+            'course_enrollment_track': 'Executive Program',
+            'course_run_enrollment_track': 'Unpaid Executive Program',
```
Comment (azanbinzahid) [inline course_discovery/apps/course_metadata/management/commands/populate_executive_education_data_csv.py]: do we need to change mgmt command name as well?

File: course_discovery/apps/course_metadata/management/commands/populate_executive_education_data_csv.py (hunk 1 above)
Comment (DawoudSheraz) [reply course_discovery/apps/course_metadata/management/commands/populate_executive_education_data_csv.py]: Message string? yes.

File: course_discovery/apps/course_metadata/management/commands/populate_executive_education_data_csv.py (hunk 3 above)
Comment (DawoudSheraz) [reply course_discovery/apps/course_metadata/management/commands/populate_executive_education_data_csv.py]: Well, technically, the courses are still ExEd.

File: course_discovery/apps/course_metadata/management/commands/populate_executive_education_data_csv.py (hunk 2 above)
Comment (DawoudSheraz) [reply course_discovery/apps/course_metadata/management/commands/populate_executive_education_data_csv.py:221]: Responded to this internally over Slack.

**Review by azanbinzahid — APPROVED** [review]
Looks good w/ current API format, please add TODO in PR that this will be updated in follow-up PRs and watch out for failing tests.

**Comment by DawoudSheraz** [comment]
The failing test on mysql8 is a known/flaky test. mysql57 tests are passing, therefore I have merged the PR.

**Pull request merged**